From ee5c0bab10e280cf54d3d2c06c6a4cc6b6ed627b Mon Sep 17 00:00:00 2001
From: Florin Sarbu <florin@balena.io>
Date: Thu, 17 Oct 2019 11:16:39 +0200
Subject: [PATCH] overlays: Add Hyperpixel4 overlays

These overlays are taken from https://github.com/pimoroni/hyperpixel4.git
They each are taken from:
  pi3                          df3f016 Merge pull request #37 from pimoroni/pi3-buster-fix
  pi4                          c958d35 Add rotate instructions to install.sh
  square                       a9c442f Updated README
  square-pi4                   eebabef Update README.md

Upstream-status: Inappropriate [not author]
Signed-off-by: Florin Sarbu <florin@balena.io>
---
 arch/arm/boot/dts/overlays/Makefile                |   4 +
 .../boot/dts/overlays/hyperpixel4-pi3-overlay.dts  |  98 ++++++++++++++++++++
 .../boot/dts/overlays/hyperpixel4-pi4-overlay.dts  |  98 ++++++++++++++++++++
 .../overlays/hyperpixel4-square-pi3-overlay.dts    | 103 +++++++++++++++++++++
 .../overlays/hyperpixel4-square-pi4-overlay.dts    | 103 +++++++++++++++++++++
 5 files changed, 406 insertions(+)
 create mode 100644 arch/arm/boot/dts/overlays/hyperpixel4-pi3-overlay.dts
 create mode 100644 arch/arm/boot/dts/overlays/hyperpixel4-pi4-overlay.dts
 create mode 100644 arch/arm/boot/dts/overlays/hyperpixel4-square-pi3-overlay.dts
 create mode 100644 arch/arm/boot/dts/overlays/hyperpixel4-square-pi4-overlay.dts

diff --git a/arch/arm/boot/dts/overlays/Makefile b/arch/arm/boot/dts/overlays/Makefile
index 87082d6..554756c 100644
--- a/arch/arm/boot/dts/overlays/Makefile
+++ b/arch/arm/boot/dts/overlays/Makefile
@@ -59,6 +59,10 @@ dtbo-$(CONFIG_ARCH_BCM2835) += \
 	hy28a.dtbo \
 	hy28b.dtbo \
 	hy28b-2017.dtbo \
+	hyperpixel4-pi3.dtbo \
+	hyperpixel4-pi4.dtbo \
+	hyperpixel4-square-pi3.dtbo \
+	hyperpixel4-square-pi4.dtbo \
 	i-sabre-q2m.dtbo \
 	i2c-bcm2708.dtbo \
 	i2c-gpio.dtbo \
diff --git a/arch/arm/boot/dts/overlays/hyperpixel4-pi3-overlay.dts b/arch/arm/boot/dts/overlays/hyperpixel4-pi3-overlay.dts
new file mode 100644
index 0000000..78bf59e
--- /dev/null
+++ b/arch/arm/boot/dts/overlays/hyperpixel4-pi3-overlay.dts
@@ -0,0 +1,98 @@
+/dts-v1/;
+/plugin/;
+/ {
+    compatible = "brcm,bcm2708";
+    fragment@0 {
+        target = <&fb>;
+        __overlay__ {
+            pinctrl-names = "default";
+            pinctrl-0 = <0x1>;
+        };
+    };
+    fragment@1 {
+        target = <&gpio>;
+        __overlay__ {
+            dpi18_pins {
+                brcm,pins = <0 1 2 3 4 5 6 7 8 9 12 13 14 15 16 17 20 21 22 23 24 25>;
+                brcm,function = <0x6>;
+                brcm,pull = <0x0>;
+                phandle = <0x1>;
+            };
+        };
+    };
+    fragment@2 {
+        target-path = "/";
+        __overlay__ {
+            rpi_backlight: rpi_backlight {
+                compatible = "gpio-backlight";
+                gpios = <&gpio 19 0>;
+                default-on;
+            };
+        };
+    };
+    fragment@3 {
+        target-path = "/";
+        __overlay__ {
+            i2c_gpio: i2c@0 {
+                compatible = "i2c-gpio";
+                gpios = <&gpio 10 0 /* sda */
+                     &gpio 11 0 /* scl */
+                    >;
+                i2c-gpio,delay-us = <4>;        /* ~100 kHz */
+                #address-cells = <1>;
+                #size-cells = <0>;
+            };
+        };
+    };
+    fragment@4 {
+        target = <&i2c_gpio>;
+        __overlay__ {
+            /* needed to avoid dtc warning */
+            #address-cells = <1>;
+            #size-cells = <0>;
+            ft6236: ft6236@5d {
+                compatible = "goodix,gt911";
+                reg = <0x5d>;
+                interrupt-parent = <&gpio>;
+                interrupts = <27 2>;
+                touchscreen-size-x = <480>;
+                touchscreen-size-y = <800>;
+                touchscreen-x-mm = <51>;
+                touchscreen-y-mm = <85>;
+            };
+        };
+    };
+    fragment@5 {
+        target = <&ft6236>;
+        __overlay__ {
+            touchscreen-swapped-x-y;
+        };
+    };
+    fragment@6 {
+        target = <&ft6236>;
+        __dormant__ {
+            touchscreen-inverted-x;
+        };
+    };
+    fragment@7 {
+        target = <&ft6236>;
+        __overlay__ {
+            touchscreen-inverted-y;
+        };
+    };
+    __symbols__ {
+        dpi18_pins = "/fragment@1/__overlay__/dpi18_pins";
+    };
+    __local_fixups__ {
+        fragment@0 {
+            __overlay__ {
+                pinctrl-0 = <0x0>;
+            };
+        };
+    };
+    __overrides__ {
+        touchscreen-inverted-x = <0>,"+6";
+        touchscreen-inverted-y = <0>,"-7";
+        touchscreen-swapped-x-y = <0>,"-5";
+    };
+};
diff --git a/arch/arm/boot/dts/overlays/hyperpixel4-pi4-overlay.dts b/arch/arm/boot/dts/overlays/hyperpixel4-pi4-overlay.dts
new file mode 100644
index 0000000..3757a99
--- /dev/null
+++ b/arch/arm/boot/dts/overlays/hyperpixel4-pi4-overlay.dts
@@ -0,0 +1,98 @@
+/dts-v1/;
+/plugin/;
+/ {
+    compatible = "brcm,bcm2708";
+    fragment@0 {
+        target = <&fb>;
+        __overlay__ {
+            pinctrl-names = "default";
+            pinctrl-0 = <0x1>;
+        };
+    };
+    fragment@1 {
+        target = <&gpio>;
+        __overlay__ {
+            dpi18_pins {
+                brcm,pins = <0 1 2 3 4 5 6 7 8 9 12 13 14 15 16 17 20 21 22 23 24 25>;
+                brcm,function = <0x6>;
+                brcm,pull = <0x0>;
+                phandle = <0x1>;
+            };
+        };
+    };
+    fragment@2 {
+        target-path = "/";
+        __overlay__ {
+            rpi_backlight: rpi_backlight {
+                compatible = "gpio-backlight";
+                gpios = <&gpio 19 0>;
+                default-on;
+            };
+        };
+    };
+    fragment@3 {
+        target-path = "/";
+        __overlay__ {
+            i2c_gpio: i2c@0 {
+                compatible = "i2c-gpio";
+                gpios = <&gpio 10 0 /* sda */
+                     &gpio 11 0 /* scl */
+                    >;
+                i2c-gpio,delay-us = <4>;        /* ~100 kHz */
+                #address-cells = <1>;
+                #size-cells = <0>;
+            };
+        };
+    };
+    fragment@4 {
+        target = <&i2c_gpio>;
+        __overlay__ {
+            /* needed to avoid dtc warning */
+            #address-cells = <1>;
+            #size-cells = <0>;
+            ft6236: ft6236@14 {
+                compatible = "goodix,gt911";
+                reg = <0x14>;
+                interrupt-parent = <&gpio>;
+                interrupts = <27 2>;
+                touchscreen-size-x = <480>;
+                touchscreen-size-y = <800>;
+                touchscreen-x-mm = <51>;
+                touchscreen-y-mm = <85>;
+            };
+        };
+    };
+    fragment@5 {
+        target = <&ft6236>;
+        __overlay__ {
+            touchscreen-swapped-x-y;
+        };
+    };
+    fragment@6 {
+        target = <&ft6236>;
+        __dormant__ {
+            touchscreen-inverted-x;
+        };
+    };
+    fragment@7 {
+        target = <&ft6236>;
+        __overlay__ {
+            touchscreen-inverted-y;
+        };
+    };
+    __symbols__ {
+        dpi18_pins = "/fragment@1/__overlay__/dpi18_pins";
+    };
+    __local_fixups__ {
+        fragment@0 {
+            __overlay__ {
+                pinctrl-0 = <0x0>;
+            };
+        };
+    };
+    __overrides__ {
+        touchscreen-inverted-x = <0>,"+6";
+        touchscreen-inverted-y = <0>,"-7";
+        touchscreen-swapped-x-y = <0>,"-5";
+    };
+};
diff --git a/arch/arm/boot/dts/overlays/hyperpixel4-square-pi3-overlay.dts b/arch/arm/boot/dts/overlays/hyperpixel4-square-pi3-overlay.dts
new file mode 100644
index 0000000..f544a6c
--- /dev/null
+++ b/arch/arm/boot/dts/overlays/hyperpixel4-square-pi3-overlay.dts
@@ -0,0 +1,103 @@
+/dts-v1/;
+/plugin/;
+/ {
+	compatible = "brcm,bcm2708";
+	fragment@0 {
+		target = <0xdeadbeef>;
+		__overlay__ {
+			pinctrl-names = "default";
+			pinctrl-0 = <0x1>;
+		};
+	};
+	fragment@1 {
+		target = <0xdeadbeef>;
+		__overlay__ {
+			dpi18_pins {
+				brcm,pins = <0x0 0x1 0x2 0x3 0x4 0x5 0x6 0x7 0x8 0x9	0xc 0xd 0xe 0xf 0x10 0x11	 0x14 0x15 0x16 0x17 0x18 0x19>;
+				brcm,function = <0x6>;
+				brcm,pull = <0x0>;
+				phandle = <0x1>;
+			};
+		};
+	};
+	fragment@2 {
+                target = <&gpio>;
+                __overlay__ {
+                        edt_ft5x06_pins: edt_ft5x06_pins {
+                                brcm,pins = <27>;
+                                brcm,function = <0>;
+                                brcm,pull = <2>;
+                        };
+                };
+	};
+	fragment@3 {
+		target-path = "/";
+		__overlay__ {
+			i2c_gpio: i2c@0 {
+				compatible = "i2c-gpio";
+				gpios = <&gpio 10 0 /* sda */
+					 &gpio 11 0 /* scl */
+					>;
+				i2c-gpio,delay-us = <1>;		/* ~100 kHz */
+				#address-cells = <1>;
+				#size-cells = <0>;
+			};
+		};
+	};
+	fragment@4 {
+		target = <&i2c_gpio>;
+                __overlay__ {
+                        #address-cells = <1>;
+                        #size-cells = <0>;
+                        status = "okay";
+
+                        polytouch: edt-ft5x06@48 {
+                                #address-cells = <1>;
+                                #size-cells = <0>;
+                                compatible = "edt,edt-ft5406", "edt,edt-ft5x06";
+                                reg = <0x48>;
+                                pinctrl-names = "default";
+                                pinctrl-0 = <&edt_ft5x06_pins>;
+                                interrupt-parent = <&gpio>;
+                                interrupts = <27 0x02>;
+                                touchscreen-size-x = <720>;
+                                touchscreen-size-y = <720>;
+                        };
+                };
+	};
+	fragment@5 {
+		target-path = "/";
+		__overlay__ {
+			rpi_backlight: rpi_backlight {
+				compatible = "gpio-backlight";
+				gpios = <&gpio 19 0>;
+				default-on;
+			};
+		};
+	};
+	fragment@6 {
+		target = <&polytouch>;
+		__overlay__ {
+			touchscreen-inverted-x = <1>;
+			touchscreen-inverted-y = <1>;
+		};
+	};
+	__symbols__ {
+		dpi18_pins = "/fragment@1/__overlay__/dpi18_pins";
+	};
+	__local_fixups__ {
+		fragment@0 {
+			__overlay__ {
+				pinctrl-0 = <0x0>;
+			};
+		};
+	};
+	__fixups__ {
+		leds = "/fragment@0:target:0";
+		gpio = "/fragment@1:target:0";
+	};
+	__overrides__ {
+		rotate = <0>,"-6";
+	};
+};
+
diff --git a/arch/arm/boot/dts/overlays/hyperpixel4-square-pi4-overlay.dts b/arch/arm/boot/dts/overlays/hyperpixel4-square-pi4-overlay.dts
new file mode 100644
index 0000000..f544a6c
--- /dev/null
+++ b/arch/arm/boot/dts/overlays/hyperpixel4-square-pi4-overlay.dts
@@ -0,0 +1,103 @@
+/dts-v1/;
+/plugin/;
+/ {
+	compatible = "brcm,bcm2708";
+	fragment@0 {
+		target = <0xdeadbeef>;
+		__overlay__ {
+			pinctrl-names = "default";
+			pinctrl-0 = <0x1>;
+		};
+	};
+	fragment@1 {
+		target = <0xdeadbeef>;
+		__overlay__ {
+			dpi18_pins {
+				brcm,pins = <0x0 0x1 0x2 0x3 0x4 0x5 0x6 0x7 0x8 0x9	0xc 0xd 0xe 0xf 0x10 0x11	 0x14 0x15 0x16 0x17 0x18 0x19>;
+				brcm,function = <0x6>;
+				brcm,pull = <0x0>;
+				phandle = <0x1>;
+			};
+		};
+	};
+	fragment@2 {
+                target = <&gpio>;
+                __overlay__ {
+                        edt_ft5x06_pins: edt_ft5x06_pins {
+                                brcm,pins = <27>;
+                                brcm,function = <0>;
+                                brcm,pull = <2>;
+                        };
+                };
+	};
+	fragment@3 {
+		target-path = "/";
+		__overlay__ {
+			i2c_gpio: i2c@0 {
+				compatible = "i2c-gpio";
+				gpios = <&gpio 10 0 /* sda */
+					 &gpio 11 0 /* scl */
+					>;
+				i2c-gpio,delay-us = <1>;		/* ~100 kHz */
+				#address-cells = <1>;
+				#size-cells = <0>;
+			};
+		};
+	};
+	fragment@4 {
+		target = <&i2c_gpio>;
+                __overlay__ {
+                        #address-cells = <1>;
+                        #size-cells = <0>;
+                        status = "okay";
+
+                        polytouch: edt-ft5x06@48 {
+                                #address-cells = <1>;
+                                #size-cells = <0>;
+                                compatible = "edt,edt-ft5406", "edt,edt-ft5x06";
+                                reg = <0x48>;
+                                pinctrl-names = "default";
+                                pinctrl-0 = <&edt_ft5x06_pins>;
+                                interrupt-parent = <&gpio>;
+                                interrupts = <27 0x02>;
+                                touchscreen-size-x = <720>;
+                                touchscreen-size-y = <720>;
+                        };
+                };
+	};
+	fragment@5 {
+		target-path = "/";
+		__overlay__ {
+			rpi_backlight: rpi_backlight {
+				compatible = "gpio-backlight";
+				gpios = <&gpio 19 0>;
+				default-on;
+			};
+		};
+	};
+	fragment@6 {
+		target = <&polytouch>;
+		__overlay__ {
+			touchscreen-inverted-x = <1>;
+			touchscreen-inverted-y = <1>;
+		};
+	};
+	__symbols__ {
+		dpi18_pins = "/fragment@1/__overlay__/dpi18_pins";
+	};
+	__local_fixups__ {
+		fragment@0 {
+			__overlay__ {
+				pinctrl-0 = <0x0>;
+			};
+		};
+	};
+	__fixups__ {
+		leds = "/fragment@0:target:0";
+		gpio = "/fragment@1:target:0";
+	};
+	__overrides__ {
+		rotate = <0>,"-6";
+	};
+};
+
-- 
2.7.4

